mono/utils/mono-threads-coop.c

```diff
@@ -231,8 +242,33 @@ mono_threads_cookie_for_reset_blocking_start (MonoThreadInfo *info, int reset_bl
 	return info;
 }
 
-void*
-mono_threads_reset_blocking_start (void* stackdata)
+gpointer
+mono_threads_reset_blocking_start (gpointer *stackdata)
+{
+	gpointer cookie;
+
+	if (!mono_threads_is_coop_enabled ())
+		return NULL;
+
+#ifdef ENABLE_CHECKED_BUILD_GC
+	int reset_blocking_count = InterlockedIncrement (&coop_reset_blocking_count) + 1;
+#endif
+
+	cookie = mono_threads_reset_blocking_start_unbalanced (stackdata);
+
+#ifdef ENABLE_CHECKED_BUILD_GC
+	if (mono_check_mode_enabled (MONO_CHECK_MODE_GC)) {
+		int level G_GNUC_UNUSED = coop_tls_push (reset_blocking_count);
+		//g_warning("Entering reset nest; level %d; cookie %d\n", level, reset_blocking_count);
+		return (void *)(intptr_t)reset_blocking_count;
+	}
```

Comment:
It would be nicer if we avoided the code duplication with `mono_threads_cookie_for_reset_blocking_start`